plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1332,7 +1332,11 @@ function ( $subscription ) use ( $site_id ) {
 	 * @return array Array of state_name => (bool) state
 	 */
 	public static function get_product_subscription_state( $product_id ) {
-		$subscription = self::_get_subscriptions_from_product_id( $product_id, true );
+		$product_subscriptions = wp_list_filter( self::get_installed_subscriptions(), array( 'product_id' => $product_id ) );
+
+		$subscription = ! empty( $product_subscriptions )
+			? array_shift( $product_subscriptions )
+			: array();
 
 		return array(
 			'unregistered' => empty( $subscription ),
```

Comment:
@gedex when there is no connected subscription `unregistered` will be `true`.  So [here](https://github.com/woocommerce/woocommerce/blob/update/show-expiration-related-modal-on-installed-subscriptions/plugins/woocommerce/includes/admin/helper/class-wc-product-usage-notice.php#L93), `self::$product_usage_notice_rules` will still have a product id with an empty state. 

So the notice will still be displayed.

Reply:
Currently it's intended.

When this line is reached, from https://github.com/woocommerce/woocommerce/blob/afc266ec869965d1c1d8daf0902e5a24373433cb/plugins/woocommerce/includes/admin/helper/class-wc-product-usage-notice.php#L298

the user is on the product screen that matches with the product ID and rule — basically the product is currently used on the site without connected / installed subscriptions are found. Possible three cases:

1. It's a legit case of no subscriptions being used
2. There's a product subscription but activated / connected on another site
3. There's a product subscription but NOT activated / connected on any site